man/sd_bus_service_reconnect.c

```diff
@@ -135,15 +153,16 @@ static int setup(object *o) {
                                  o));
   /* By default the service is only assigned an ephemeral name. Also add a well-known
    * one, so that clients know whom to call. This needs to be asynchronous, as
-   * D-Bus might not be yet available.
+   * D-Bus might not be yet available. The callback will check whether the error is
+   * expected or not, in case it fails.
    * https://www.freedesktop.org/software/systemd/man/sd_bus_request_name.html
    */
   check(sd_bus_request_name_async(*o->bus,
                                   NULL,
                                   "org.freedesktop.ReconnectExample",
                                   0,
-                                  NULL,
-                                  NULL));
+                                  request_name_callback,
+                                  o));
```

Comment:
hmm, frankly, this should be moved into a signal handler that watches for the Connected signal.